src/tree.c

```diff
@@ -135,7 +135,7 @@ static git_tree_entry *alloc_entry(const char *filename)
 
 struct tree_key_search {
 	const char *filename;
-	uint16_t filename_len;
+	size_t filename_len;
```

Comment:
This makes the `filename_len` field a different size for the field in the `git_tree_entry` which I'd rather avoid.